yarn-project/archiver/src/archiver/l1_to_l2_message_store.ts

```diff
@@ -16,21 +16,24 @@ export class L1ToL2MessageStore {
 
   addMessage(messageKey: Fr, msg: L1ToL2Message) {
     const messageKeyBigInt = messageKey.value;
-    if (this.store.has(messageKeyBigInt)) {
-      this.store.get(messageKeyBigInt)!.count++;
+    const msgAndCount = this.store.get(messageKeyBigInt);
+    if (msgAndCount) {
+      msgAndCount.count++;
+      this.store.set(messageKeyBigInt, msgAndCount);
     } else {
       this.store.set(messageKeyBigInt, { message: msg, count: 1 });
     }
   }
 
   removeMessage(messageKey: Fr) {
     const messageKeyBigInt = messageKey.value;
-    if (!this.store.has(messageKeyBigInt)) {
+    const msgAndCount = this.store.get(messageKeyBigInt);
+    if (!msgAndCount) {
       return;
     }
-    const count = this.store.get(messageKeyBigInt)!.count;
-    if (count > 1) {
-      this.store.get(messageKeyBigInt)!.count--;
+    if (msgAndCount.count > 1) {
+      msgAndCount.count--;
+      this.store.set(messageKeyBigInt, msgAndCount);
```

Comment:
I think msgAndCount returns a reference to the item so you dont actually need to store it again?
```
Returns a specified element from the Map object. If the value that is associated to the provided key is an object, then you will get a reference to that object and any change made to that object will effectively modify it inside the Map.
```
from the js doc, same applies to above change